File: python/petshop.py
class PetShop:
    # konstruktor untuk menginisialisasi atribut produk
    def __init__(self, id_produk, nama_produk, kategori_produk, harga_produk):
        self.id_produk = id_produk
        self.nama_produk = nama_produk
        self.kategori_produk = kategori_produk
        self.harga_produk = harga_produk

    # metode untuk mendapatkan id produk
    def get_id_produk(self):
        return self.id_produk

    # metode untuk mendapatkan nama produk
    def get_nama_produk(self):
        return self.nama_produk

    # metode untuk mendapatkan kategori produk
    def get_kategori_produk(self):
        return self.kategori_produk

    # metode untuk mendapatkan harga produk
    def get_harga_produk(self):
        return self.harga_produk

    # metode untuk mengatur id produk
    def set_id_produk(self, id_produk):
        self.id_produk = id_produk

    # metode untuk mengatur nama produk
    def set_nama_produk(self, nama_produk):
        self.nama_produk = nama_produk

    # metode untuk mengatur kategori produk
    def set_kategori_produk(self, kategori_produk):
        self.kategori_produk = kategori_produk

    # metode untuk mengatur harga produk
    def set_harga_produk(self, harga_produk):
        self.harga_produk = harga_produk

    # metode untuk membuat produk baru
    def create_produk(self, id_produk, nama_produk, kategori_produk, harga_produk):
        self.id_produk = id_produk
        self.nama_produk = nama_produk
        self.kategori_produk = kategori_produk
        self.harga_produk = harga_produk

    # metode untuk membaca informasi produk
    def read_produk(self):
        print(f"ID Produk: {self.id_produk}")
        print(f"Nama Produk: {self.nama_produk}")
        print(f"Kategori Produk: {self.kategori_produk}")
        print(f"Harga Produk: {self.harga_produk}")

    # metode untuk memperbarui informasi produk
    def update_produk(self, id_produk, nama_produk, kategori_produk, harga_produk):
        self.id_produk = id_produk
        self.nama_produk = nama_produk
        self.kategori_produk = kategori_produk
        self.harga_produk = harga_produk

    # metode untuk menghapus produk
    def delete_produk(self):
        self.id_produk = 0
        self.nama_produk = ""
        self.kategori_produk = ""
        self.harga_produk = 0
        
    # destruktor untuk menghapus objek produk
    def __del__(self):
        print(f"Produk dengan ID {self.id_produk} telah dihapus.")

File: java/PetShop.java
public class PetShop {
    // deklarasi atribut kelas
    private int idProduk;
    private String namaProduk;
    private String kategoriProduk;
    private int hargaProduk;

    // konstruktor untuk inisialisasi objek
    public PetShop(int idProduk, String namaProduk, String kategoriProduk, int hargaProduk) {
        this.idProduk = idProduk;
        this.namaProduk = namaProduk;
        this.kategoriProduk = kategoriProduk;
        this.hargaProduk = hargaProduk;
    }

    // getter untuk mendapatkan id produk
    public int getIdProduk() {
        return idProduk;
    }

    // getter untuk mendapatkan nama produk
    public String getNamaProduk() {
        return namaProduk;
    }

    // getter untuk mendapatkan kategori produk
    public String getKategoriProduk() {
        return kategoriProduk;
    }

    // getter untuk mendapatkan harga produk
    public int getHargaProduk() {
        return hargaProduk;
    }

    // setter untuk mengubah id produk
    public void setIdProduk(int idProduk) {
        this.idProduk = idProduk;
    }

    // setter untuk mengubah nama produk
    public void setNamaProduk(String namaProduk) {
        this.namaProduk = namaProduk;
    }

    // setter untuk mengubah kategori produk
    public void setKategoriProduk(String kategoriProduk) {
        this.kategoriProduk = kategoriProduk;
    }

    // setter untuk mengubah harga produk
    public void setHargaProduk(int hargaProduk) {
        this.hargaProduk = hargaProduk;
    }

    // method untuk membuat produk baru
    public void createProduk(int idProduk, String namaProduk, String kategoriProduk, int hargaProduk) {
        setIdProduk(idProduk);
        setNamaProduk(namaProduk);
        setKategoriProduk(kategoriProduk);
        setHargaProduk(hargaProduk);
    }

    // method untuk membaca informasi produk
    public void readProduk() {
        System.out.println("ID Produk: " + getIdProduk());
        System.out.println("Nama Produk: " + getNamaProduk());
        System.out.println("Kategori Produk: " + getKategoriProduk());
        System.out.println("Harga Produk: " + getHargaProduk());
    }

    // method untuk mengupdate informasi produk
    public void updateProduk(int idProduk, String namaProduk, String kategoriProduk, int hargaProduk) {
        setIdProduk(idProduk);
        setNamaProduk(namaProduk);
        setKategoriProduk(kategoriProduk);
        setHargaProduk(hargaProduk);
    }

    // method untuk menghapus informasi produk
    public void deleteProduk() {
        setIdProduk(0);
        setNamaProduk("");
        setKategoriProduk("");
        setHargaProduk(0);
    }
}
